### DISCUSSION TITLE
OR and WHERE on Eager Loading always ended with "missing FROM-clause entry"

### DISCUSSION THREAD

**Discussion by elazsyahranie** [sequelize/sequelize#15638] [Help & Questions] [answered]
Below is the structure of my Eager Loading 
`
    const handler = await mainModel.findAll({

      include: [
        nestedModel,
        nestedModel2,
        nestedModel3,
        nestedModel4,
        nestedModel5,
        nestedModel6
      ],
      where: where,
      limit: 3,
    });
` 

The `where` is an Object with structure below: 

`
{
  [Op.or]: [
    { 
      column_one: { [Op.iLike]: %${req.query.keywords}% } 
    }, 
    {
       '$NestedModel4->NestedNestedModel.name$': { [Op.iLike]: %${req.query.keywords}% }
    }
  ]
}
` 

Now the issue is, I'm about to make a filter where, if I put some characters in the `search` query, it would do something like 

`WHERE mainModel. name ILIKE '%Hello%' 
OR public.mskills.name ILIKE '%A%'` 

When I applied the query above, which I made myself, on pgAdmin, it worked as I expected. But, when I try to apply the code above on Sequelize, it always return `"missing FROM-clause entry for table \"NestedModel4->NestedNestedModel.name\""` 

Can anybody show where I made mistake? For the record, I followed [Sequelize documentation](https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#complex-where-clauses-at-the-top-level)

**Comment by ephys** [comment, marked as answer]
`$NestedModel4->NestedNestedModel.name$` is not a valid attribute selector, it's `$NestedModel4.NestedNestedModel.name$`

**Reply by elazsyahranie** [reply]
In my database, the "NestedNestedModel" is named as (I can't show the real name because this is my company's project) "NestedTableNames". In the attribute selector, it seems that it has to be written without plural. So instead of

`$NestedModel4.NestedNestedModels.name$` 

it should be 

`$NestedModel4.NestedNestedModel.name$` 

Am I right?

**Reply by ephys** [reply]
It's not the name of the model but the name of the association, you can `console.log` the `association` static property of your model to see what your associations are named
